src/utils.ts

```diff
@@ -51,24 +49,34 @@ type ReturnJsonObject = { [Key in string]?: ReturnJsonValue; };
 type ReturnJsonArray = Array<ReturnJsonValue>;
 
 /**
- * Helper function that traverses JSON structure and parses fields such as modifiedAt or createdAt to dates.
+ * Traverses JSON structure and converts fields that end with "At" to a Date object (fields such as "modifiedAt" or
+ * "createdAt").
+ *
+ * If you want parse other fields as well, you can provide a custom matcher function shouldParseField(). This
+ * admittedly awkward approach allows this function to be reused for various purposes without introducing potential
+ * breaking changes.
+ *
+ * If the field cannot be converted to Date, it is left as is.
  */
-export function parseDateFields(input: JsonValue, depth = 0): ReturnJsonValue {
+export function parseDateFields(input: JsonValue, shouldParseField: ((key: string) => boolean) | null = null, depth = 0): ReturnJsonValue {
+    const PARSE_DATE_FIELDS_KEY_SUFFIX = 'At';
+    const PARSE_DATE_FIELDS_MAX_DEPTH = 3; // obj.data.someArrayField.[x].field
```

Comment:
(nit) I think we rarely put const definitions inside the function like this. Generally, the consts are named in a way that makes it clear for what they're meant and it's easier to reason about them when they're all defined together at the top of the file.

Reply:
Here is my reasoning:

* I find it useful for the method's documentation (or implementation for that matter) to be straightforward about what it does
* That means explicitly mentioning the "At" suffix
* In order to keep the code in sync (should that ever change), all the mentions should be as close to each other possible

Moreover:
* Both constants are really only used here
* It is not configuration in the true sense

So, I decided to just inline the values 😅

Alternatives:
* Rename to `parseDateFieldsKeySuffix` (snake case should be less "triggering")
* Move `parseDateFields` to a standalone file. Then the constants can stay as they are as they will be clearly scoped.

Reply:
inlining is ok